packages/protocol/contracts/L2/Lib1559Math.sol

```diff
@@ -61,8 +61,15 @@ library Lib1559Math {
         int256 lnRatio = FixedPointMathLib.lnWad(int256(ratio)); // may be negative
 
         uint256 newGasExcess;
+
         assembly {
-            newGasExcess := sdiv(add(mul(lnRatio, _newGasTarget), mul(ratio, _gasExcess)), f)
+            // x = (_newGasTarget * lnRatio + _gasExcess * ratio)
+            let x := add(mul(_newGasTarget, lnRatio), mul(_gasExcess, ratio))
+
+            // If x < 0, set newGasExcess to 0, otherwise calculate newGasExcess = x / f
```

Comment:
If it can still be 0, maybe easier to just accept and just always set it back to 0?

Reply:
Not sure what you meant, but x (the new gas excess) can be zero. Here we need to set it to 0 if it is negative before we perform `x/f`.

Reply:
Oh I just mean that the math to calculate the `newGasExcess` is quite complex, and does not seem to guarantee that we can calculate the correct value to maintain the current basefee in all cases, so if it's still sometimes reset to 0 then I guess okay to always do it to make the code simpler in general?

Reply:
I see. If the math is even harder to implement and test, I would agree to simply set the gasExcess to 0, but since the math has ben actually figured out, I think we should still maintain base fee unchanged (or slightly changed) when configurations changed.